ksqldb-engine/src/main/java/io/confluent/ksql/ddl/commands/DdlCommandExec.java

```diff
@@ -81,6 +81,16 @@ private Executor(
 
     @Override
     public DdlCommandResult executeCreateStream(final CreateStreamCommand createStream) {
+      final SourceName sourceName = createStream.getSourceName();
+      final DataSource dataSource = metaStore.getSource(sourceName);
+
+      if (dataSource != null && !createStream.isOrReplace()) {
+        return new DdlCommandResult(true,
+            String.format("Cannot add stream %s: A stream with the same name "
```

Comment:
nit: i think we should use `dataSource.getDataSourceType().getKsqlType()` here and in `executeCreateTable`

Reply:
Thanks. I updated the PR.